tests-e2e/fixture/fixture.go

```diff
@@ -0,0 +1,196 @@
+	if err != nil && !apierr.IsNotFound(err) {
+		return err
+	}
+
+	// Wait for namespace to delete
+	if err := wait.Poll(time.Second*1, time.Minute*2, func() (done bool, err error) {
+
+		_, err = kubeClientSet.CoreV1().Namespaces().Get(context.Background(), GitOpsServiceE2ENamespace, metav1.GetOptions{})
+		if err != nil {
+			if apierr.IsNotFound(err) {
+				return true, nil
+			} else {
+				return false, err
+			}
+		}
+
+		return false, nil
+	}); err != nil {
+		return err
+	}
+
+	// Create the namespace again
+	_, err = kubeClientSet.CoreV1().Namespaces().Create(context.Background(), &corev1.Namespace{ObjectMeta: metav1.ObjectMeta{
+		Name: GitOpsServiceE2ENamespace,
+		Labels: map[string]string{
+			"argocd.argoproj.io/managed-by": "gitops-service-argocd",
+		},
+	}}, metav1.CreateOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Wait for Argo CD to process the namespace, before we exit:
+	// - This helps us avoid a race condition where the namespace is created, but Argo CD has not yet
+	//   set up proper roles for it.
+
+	if err := wait.Poll(time.Second*1, time.Minute*2, func() (done bool, err error) {
+		var roleBindings *v1.RoleBindingList
+		roleBindings, err = kubeClientSet.RbacV1().RoleBindings(GitOpsServiceE2ENamespace).List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		count := 0
+		// Exit the poll loop when there exists at least one rolebinding containing 'gitops-service-argocd' in it's name
+		// This helps us avoid a race condition where the namespace is created, but Argo CD has not yet
+		// set up proper roles for it.
+		for _, item := range roleBindings.Items {
+			if strings.Contains(item.Name, "gitops-service-argocd-") {
+				count++
+			}
+		}
+
+		if count >= 2 {
+			// We expect at least 2 rolebindings
+			return true, nil
+		}
+
+		return false, nil
+	}); err != nil {
+		return err
+	}
+
+	if err := wait.Poll(time.Second*1, time.Minute*2, func() (done bool, err error) {
+		var roles *v1.RoleList
+		roles, err = kubeClientSet.RbacV1().Roles(GitOpsServiceE2ENamespace).List(context.Background(), metav1.ListOptions{})
+		if err != nil {
+			return false, err
+		}
+
+		count := 0
+		// Exit the poll loop when there exists at least one rolebinding containing 'gitops-service-argocd' in it's name
+		// This helps us avoid a race condition where the namespace is created, but Argo CD has not yet
+		// set up proper roles for it.
+		for _, item := range roles.Items {
+			if strings.Contains(item.Name, "gitops-service-argocd-") {
+				count++
+			}
+		}
+
+		if count >= 2 {
+			// We expect at least 2 roles
+			return true, nil
+		}
+
+		return false, nil
+	}); err != nil {
+		return err
+	}
+
+	return nil
+
+}
+
+func GetKubeConfig() (*rest.Config, error) {
+
+	overrides := clientcmd.ConfigOverrides{}
+
+	loadingRules := clientcmd.NewDefaultClientConfigLoadingRules()
+	clientConfig := clientcmd.NewInteractiveDeferredLoadingClientConfig(loadingRules, &overrides, os.Stdin)
+
```

Comment:
I tried to add this one, but the logging was way too verbose: it output a 'getting the kubeconfig' message every second (because the existing fixtures are retrieving the client at the start of every attempt).